src/payment/data/reducers.js

```diff
@@ -24,6 +25,12 @@ const basket = (state = basketInitialState, action = null) => {
   switch (action.type) {
     case BASKET_DATA_RECEIVED: return { ...state, ...action.payload };
 
+    // For submission, we only really need to know whether it's actively submitting or not.
+    case submitPayment.TRIGGER: return { ...state, submitting: true };
+    case submitPayment.SUCCESS: return { ...state, submitting: false };
```

Comment:
Since is triggered on success or failure, we don't need submitting false for the success of failure actions.

Reply:
Not sure I follow... `TRIGGER` happens at the beginning, `SUCCESS` and `FAILURE` are ending conditions so we need to set `submitting` back to false since we're done.